File: SuffleIDE/Controller/Controller.fs
module Suffle.Controller.Controller

open Suffle.Interpreter.Interpreter
open Suffle.Parser
open System.IO

[<EntryPoint>]
let main argv = 
    if argv.Length = 1 then
        let program = File.ReadAllText argv.[0]
        let ast = parse program
        let interpreter = new Suffle.Interpreter.Expression.Interpreter()
        printf "%s" <| interpreter.EvaluateProgram ast
    0 // возвращение целочисленного кода выхода

File: SuffleIDE/InterpreterSIDE/Controller.cs
using Suffle.Interpreter;
using Parser;

using Side.Interfaces.Services;

namespace InterpreterSIDE
{
    public class Controller : IInterpreter
    {
        public string Interpret(string code)
        {
            var inter = new Suffle.Interpreter.Expression.Interpreter();
            return inter.EvaluateProgram(Suffle.Parser.parse(code));
        }
    }
}
